### PR TITLE
Bug/guard pools in swap

### PR DESCRIPTION
**Pull request by ryardley**
This PR guards against the following error:

image

* Fixes issue in `CompositePool` where `contains()` method was unusable as was returning incorrect result.
* Guards against pools trying to swap assets they don't contain until eventual consistency is reached.
* Caches contains lookup in computed value
* Make it so swapCalculator simply returns `priceImpact` and `providerFee` to avoid duplication and as they are really only computed properties based on inputs - this will be easier to refactor to a render function later too.
* Add store accessor to debug store in browser

image

### CODE REVIEW

**Review by 59023g — DISMISSED** [review]
Looks good. I propose we wait a few hours until @iverc puts his work in before this is merged. Just to keep his runway clear. Looks like there's some overlap

**Comment by ryardley** [comment]
> Looks good. I propose we wait a few hours until @iverc puts his work in before this is merged. Just to keep his runway clear. Looks like there's some overlap

Been thinking the same myself

**Review by 59023g — APPROVED** [review]
Looks good. Curious of @iverc thoughts here

File: ui/app/src/hooks/useCore.ts [hunk 1]
```diff
@@ -17,6 +17,13 @@ const store = createStore();
 const actions = createActions({ store, api });
 const poolFinder = createPoolFinder(store);
 
+// expose store on the window so it is easy to inspect
+Object.defineProperty(window, "store", {
```
Comment (iverc) [inline ui/app/src/hooks/useCore.ts:21]: Should we remove this part when done testing?

Comment (ryardley) [reply ui/app/src/hooks/useCore.ts:21]: I think it is useful to be able to inspect this generally and will help with isolating some of these UI bugs. There should be no sensitive information here and you cannot write to the store.

Comment (59023g) [reply ui/app/src/hooks/useCore.ts:21]: I agree. Let's keep for now

Comment (ryardley) [reply ui/app/src/hooks/useCore.ts:21]: By the way I tried exposing it for Vue devtools on the App component but that chugged big time and this seemed to be the better way to go.

**Review by iverc — APPROVED** [review]

**Review by 59023g — APPROVED** [review]
💯

**Pull request merged**